server/cars.js

```diff
@@ -29,6 +32,44 @@ const carSchema = new mongoose.Schema({
 
 const Car = mongoose.model('Car', carSchema);
 
+
+//--------HELPER FUNCIONS--------//
+/**
+ * Returns the mileage the service is next due based on 
+ * when it was last done and the interval
+ * @param {number} currentMileage - the current mileage of the car
+ * @param {number} interval - how often the service is done (miles)
+ * @param {number} lastDone - the mileage the service was last done
+ */
+const getNextService = (currentMileage, interval, lastDone) => {
+  // If user didn't provide the last service date return the 
+  // next closest interval after current mileage. i.e. if current 
+  // mileage is 8,000 and interval is 5,000 return 10,000
+  if (!lastDone) {
+    const milesTilNext = interval - (currentMileage % interval)
+    return currentMileage + milesTilNext
+  }
+
+  // Next service is the interval since last done
+  return lastDone + interval
+}
+
+/**
+ * Based on when the car was registered and average miles driver per year,
+ * returns the estimated current mileage on the car
+ * @param {*} carMileage - the mileage of the car when it was registered
+ * @param {number} milesPerYear - miles driven in the car each year
+ * @param {*} registerDate - the date the car was registered with us
+ */
+const calculateCurrentMileage = (carMileage, milesPerYear, registerDate) => {
```

Comment:
I added `estimatedCurrentMileage` to the object returned from getCar. This function calculates that value